src/test/java/jenkins/plugins/collabnet/steps/TestPublishEventQStep.java

```diff
@@ -0,0 +1,41 @@
+package jenkins.plugins.collabnet.steps;
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import hudson.model.Result;
+import jenkins.plugins.collabnet.steps.PublishEventQStep;
+
+public class TestPublishEventQStep {
+    
+    @Rule
+    public JenkinsRule jenkins = new JenkinsRule();
+
+    @Test
+    public void buildWithEmptyServerUrlMustFail() throws Exception {
+        WorkflowJob p = jenkins.jenkins.createProject(WorkflowJob.class, "p");
+        
+        p.setDefinition(new CpsFlowDefinition(
+                "publishEventQ serverUrl: '', sourceKey: '1234', markUnstable: true"
+        ));
+        WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+        jenkins.assertBuildStatus(Result.UNSTABLE, jenkins.waitForCompletion(b1));
+        jenkins.assertLogContains(PublishEventQStep.PublishEventQStepExecution.LOG_MESSAGE_INVALID_URL, b1);
+    }
+
+    @Test
+    public void buildWithEmptySourceKeyMustFail() throws Exception {
+        WorkflowJob p = jenkins.jenkins.createProject(WorkflowJob.class, "p");
+        
+        p.setDefinition(new CpsFlowDefinition(
+                "publishEventQ serverUrl: 'https://ctf.open.collab.net', sourceKey: '', markUnstable: true"
```

Comment:
Yes, it is whatever the orchestrate client (in the existing plugin) needs. I can add validation, but didn't see it in the existing notifier code.